### PR TITLE
feat(UserProp): Add required attribute

### PR DESCRIPTION
**Pull request by okonet**
Add required and defaultValue examples.

This is a follow up for CSF work needed in https://github.com/webstudio-is/webstudio-designer/pull/53

### CODE REVIEW

**Review by kof — APPROVED** [review]
👍

File: src/components/button.tsx [hunk 1]
```diff
@@ -2,9 +2,14 @@ import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
...
+
 export const Button = forwardRef<
   ElementRef<typeof defaultTag>,
-  ComponentProps<typeof defaultTag>
->((props, ref) => <button {...props} ref={ref} />);
+    ButtonProps
+>((props, ref) => <button style={{border:"none"}} {...props} ref={ref} />);
 
+Button.defaultProps = {
+    type: "button", // otherwise it will be submit
```
Comment (kof) [inline src/components/button.tsx]: interesting, should we keep it submit since its the browser default?

Comment (okonet) [reply src/components/button.tsx]: Normally I’d say no but if we thinking about defaults then it’s more aligned yeah

File: src/components/button.tsx [hunk 2]
```diff
@@ -2,9 +2,14 @@ import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
...
 const defaultTag = "button";
 
+export type ButtonProps = ComponentProps<typeof defaultTag>
+
 export const Button = forwardRef<
   ElementRef<typeof defaultTag>,
-  ComponentProps<typeof defaultTag>
->((props, ref) => <button {...props} ref={ref} />);
+    ButtonProps
+>((props, ref) => <button style={{border:"none"}} {...props} ref={ref} />);
```
Comment (kof) [inline src/components/button.tsx]: so far things like that went into default style at the definition of the primitive  component

https://github.com/webstudio-is/webstudio-designer/blob/main/app/shared/canvas-components/primitives/box.tsx#L5

but we need to think carefully again about this, probably need to merge designer primitives with the SDK

Comment (okonet) [reply src/components/button.tsx]: This should not be there. It was a debug code so a good catch!

**Review by kof — APPROVED** [review]

**Pull request merged**